### PR TITLE
feat(dart/catalyst_cardano_serialization): add support for additional transaction body fields

### PR DESCRIPTION
**Pull request by ilap**
# Description

This pull request adds initial support for additional transaction body fields to the `TransactionBuilder` class in the `catalyst_cardano_serialization` package.
## Related Issue(s)

Refers #710 

## Description of Changes

The following fields are added:
- validity interval start
- mint
- script data hash
- collateral inputs
- collateral return
- total collateral
- reference inputs

These fields are essential for handling more complex transactions. 
The changes are reflected in the `README.md` file and the `transaction_builder.dart` file.

## Breaking Changes

The pull request does not introduce any breaking changes.

## Please confirm the following checks

* [x] My code follows the style guidelines of this project
* [x] I have performed a self-review of my code
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation
* [x] My changes generate no new warnings
* [x] I have added tests that prove my fix is effective or that my feature works
* [x] New and existing unit tests pass locally with my changes
* [x] Any dependent changes have been merged and published in downstream module

### CODE REVIEW

**Comment by dtscalac** [comment]
@ilap Thanks for the contribution. 

Can you also add the fields you added to TransactionBuilder to the _buildBody() method? The goal is that builder collects all the necessary fields and constructs a TransactionBody from them. It seems we aren't passing the fields there.

**Comment by ilap** [comment]
> Can you also add the fields you added to TransactionBuilder to the _buildBody() method? The goal is that builder collects all the necessary fields and constructs a TransactionBody from them. It seems we aren't passing the fields there.
 done

File: catalyst_voices_packages/catalyst_cardano_serialization/lib/src/builders/transaction_builder.dart [hunk 1]
```diff
@@ -38,13 +38,34 @@ final class TransactionBuilder extends Equatable {
...
+  final MultiAsset? mint;
+
+  /// The transaction metadata as a list of key-value pairs (a map).
+  final ScriptData? scriptData;
+
+  /// Collateral inputs as nonempty set.
+  final Set<TransactionInput>? collateralInputs;
+
+  /// The list of public key hashes of addresses that are required to sign the
+  /// transaction. A nonempty set of addr keyhashes.
```
Comment (dtscalac) [inline catalyst_voices_packages/catalyst_cardano_serialization/lib/src/builders/transaction_builder.dart:54]: cspell throws an error on `keyhashes`, can we change it to:

```suggestion
  /// transaction. A nonempty set of addr key hashes.
```

**Comment by stevenj** [comment]
Thank you for the great contribution.

**Pull request merged**